src/journal/journald-server.c

```diff
@@ -252,13 +252,11 @@ void server_space_usage_message(Server *s, JournalStorage *storage) {
 
 static void server_add_acls(JournalFile *f, uid_t uid) {
         assert(f);
+        assert(uid > 0);
```

Comment:
i am very much not convinced of this commit. now you have to different checks that have very similar effect, except they are not really the same. i.e. the "uid > 0" check and the "!uid_for_system_journal(uid)" check. that's just very confusing.

let's systematically use the same check, and in particular let's use the uid_for_system_journal() name, since it makes he checks immediately understandable, since it says on the tin what we are checking here.

sorry but this commit here makes the code harder to read at no benefit.

Reply:
The aim of this commit is to find whether we need to send the message to the system journal just once.

Currently this is done in multiple places which might be a problem if the involved logic is more than just checking the (static) UID ranges. IOW we might need to do more operations in the future (i.e. if we opt for your `log-split-mode` idea, btw I'm still not sure how this could work, see my last question in the relevant thread) and we might even want to cache the result.

This change would make this easier I think.

Reply:
i think it's fine to make the checks multiple times as long as it is the same function that makes the check. you are adding distinct checks to various places now, which is worse in my pov

Reply:
sorry what do you mean by "distinct checks to various places"?

The patch is only replacing "uid_for_system_journal()" with `uid > 0` mainly because `uid_for_system_journal()`  may be an heavier operation that we want to do just once.

Reply:
uid_for_system_journal() is not doing anything complex, it's not going to be noticable slower than uid > 0. 

But yeah I think we should use the same check everywhere, there's no point in trying to optimize here. It's the wrong place and makes the code more confusing, not easier to read.

We care about readability/unformity/maintainability a lot more than mere speed actually, and in particular in a case like this where the speed effect is unproven and afaics highly unlikely to be measurable.